components/form/Input.js

```diff
@@ -1,9 +1,20 @@
-import React from 'react';
+import React, { useState } from 'react';
 import PropTypes from 'prop-types';
+import { PasswordCheckIcon } from '../../assets/svgs/svgs';
+
+const Input = ({ className, Icon, label, name, placeholder, onClick, type, register, errors }) => {
+  const [showPassword, setShowPassword] = useState(false);
+
+  const handleShowPassword = () => {
+    setShowPassword((show) => !show);
+  };
+
+  const HasIcon = `${(<PasswordCheckIcon width="18px" height="18px" fill="#292D32" onClick={handleShowPassword} />)}`;
```

Comment:
Not sure why you are not using the component directly:

```suggestion
const hasIcon = <PasswordCheckIcon width="18px" height="18px" fill="#292D32" onClick={handleShowPassword} />
```

Since you are attaching the Password Icon to the `HasIcon` variable,  it is difficult to understand your thought process here.